tests/neo.UnitTests/Consensus/UT_Consensus.cs

```diff
@@ -100,35 +98,234 @@ public void ConsensusService_Primary_Sends_PrepareRequest_After_OnStart()
                     MerkleRoot = header.MerkleRoot,
                     Timestamp = header.Timestamp,
                     Index = header.Index,
-                    NextConsensus = header.NextConsensus
+                    NextConsensus = header.NextConsensus,
+                    Transactions = new Transaction[0]
                 }
-            });
-
+            };
+            Console.WriteLine("\n==========================");
+            Console.WriteLine("Telling a new block to actor consensus...");
+            Console.WriteLine("will trigger OnPersistCompleted without OnStart flag!");
             // OnPersist will not launch timer, we need OnStart
+            actorConsensus.Tell(testPersistCompleted);
+            Console.WriteLine("\n==========================");
 
+            Console.WriteLine("\n==========================");
             Console.WriteLine("will start consensus!");
             actorConsensus.Tell(new ConsensusService.Start
             {
                 IgnoreRecoveryLogs = true
             });
 
-            Console.WriteLine("OnTimer should expire!");
-            Console.WriteLine("Waiting for subscriber message!");
-            // Timer should expire in one second (block_received_time at :01, initialized at :02)
-
-            var answer = subscriber.ExpectMsg<LocalNode.SendDirectly>();
-            Console.WriteLine($"MESSAGE 1: {answer}");
-            //var answer2 = subscriber.ExpectMsg<LocalNode.SendDirectly>(); // expects to fail!
-
+            Console.WriteLine("Waiting for subscriber recovery message...");
+            var askingForInitialRecovery = subscriber.ExpectMsg<LocalNode.SendDirectly>();
+            Console.WriteLine($"Recovery Message I: {askingForInitialRecovery}");
+            ConsensusPayload initialRecoveryPayload = (ConsensusPayload)askingForInitialRecovery.Inventory;
+            RecoveryRequest rrm = (RecoveryRequest)initialRecoveryPayload.ConsensusMessage;
```

Comment:
Hi @vncoelho , can you add some comments to these steps? It will be hard to maintain if we don't know exactly what it is trying to achieve

Reply:
I agree.

I will add some comments on the first time the procedure is done. 
The other ones will follow the same logic.

Reply:
Done.